src/rest.js

```diff
@@ -3,10 +3,30 @@ import cookies from 'js-cookie';
 import { stringify } from 'qs';
 import Vue from 'vue';
 
+const GirderTokenLength = 64;
+export const OauthTokenPrefix = '#girderToken=';
+export const OauthTokenSuffix = '__';
+
 function setCookieFromAuth(auth) {
   cookies.set('girderToken', auth.token, { expires: new Date(auth.expires) });
 }
 
+/**
+ * set cookie if special string is found in the hash.
+ * @param {Location} location
+ */
+function setCookieFromHash(location) {
+  const arr = location.hash.split(OauthTokenPrefix);
+  const token = arr[arr.length - 1].split(OauthTokenSuffix)[0];
+  if (token.length === GirderTokenLength) {
+    const expires = new Date();
+    expires.setDate((new Date()).getDate() + 7);
```

Comment:
Don't have a way to get the expiration time of this token, so make it 7 days.  Could potentially add this as another format string to girder?

Reply:
We should set it for a long time IMO; if it expires on the server, the presence of the cookie won't cause any problems, it'll just get rewritten once they log in again.

Reply:
Made it a nice round year.